Network unit test:Fix flakey Ledger test

hewison-chris opened the pull request:
Sometimes this test was failing due to not all `txs` being included in
the block that was externalized. The compare of expected and actual txs
caused an assert in the test. It now waits for all the nodes to have
the txs before initializing the block being externalized.

hewison-chris committed:
Network unit test:Fix flakey Ledger test

Sometimes this test was failing due to not all `txs` being included in
the block that was externalized. The compare of expected and actual txs
caused an assert in the test. It now waits for all the nodes to have
the txs before initializing the block being externalized.

linked0 reviewed (approved):
LGTM

Geod24 commented on source/agora/test/Ledger.d:
```diff
@@ -54,15 +54,21 @@ unittest
 
-        // send it to one node
-        txs.each!(tx => node_1.putTransaction(tx));
+        txs.each!((tx)
+        {
+            // send it to one node
+            node_1.putTransaction(tx);
+            // wait for all nodes get the txs
+            nodes.each!(node => node.hasTransactionHash(hashFull(tx)).retryFor(2.seconds));
+        });
```
```suggestion
        txs.each!(tx => node_1.putTransaction(tx));
        // wait for all nodes get the txs
        txs.each!(tx =>
            nodes.each!(node => node.hasTransactionHash(hashFull(tx)).retryFor(2.seconds))
        );
```

Instead of sending the tx, then waiting for all nodes, send all txs, then wait for all nodes ?

hewison-chris replied:
In some other tests that is what we do. Just thought I would mix it up a bit. Although I suppose we test gossiping in another test.

hewison-chris replied:
Just re-read your comment and realized what you meant. Yes it would be more efficient. Will update.

hewison-chris replied:
Updated

kchulj commented:
Does this affect the faucet?

hewison-chris commented:
> Does this affect the faucet?

No I was just getting rid of one of the spurious failures in unit tests.

kchulj reviewed (approved):
LGTM

hewison-chris commented:
I see the VariableBlockSize test has the same problem. Will add a commit for that too.

hewison-chris committed:
Network unit tests:Fix a few more flakey tests

A few more tests also send more than one `tx` before creating a block.
To ensure the same number of `txs` at each node the tests either only
send one `tx` or send to all nodes before externalizing the block.

hewison-chris commented:
Found a couple more and updated them also.
